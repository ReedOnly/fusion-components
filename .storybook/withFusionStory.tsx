import React from 'react';
import Header from '../src/components/core/Header';
import Content from '../src/components/core/Content';
import Overlay from '../src/components/general/ApplicationGuidance/components/Overlay';

export default (title: string, padding: number = 16) => (stories: () => React.ReactNode) => {
    return (
        <>
            <Overlay fixed>
                <Header aside={null} start={null} content={null} />
            </Overlay>
            <Content>
                <Overlay>
                    <div
                        style={{
                            padding,
                            minHeight: '100%',
                            width: '100%',
                            maxWidth: '100%',
                            boxSizing: 'border-box',
                        }}
                    >
                        {stories()}
                    </div>
                </Overlay>
            </Content>
        </>
    );
};
