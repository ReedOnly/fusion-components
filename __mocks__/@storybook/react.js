import renderer from 'react-test-renderer';
import addons, { mockChannel } from '@storybook/addons';
import { withFusionContext } from '../../.storybook/withFusionContext';

addons.setChannel(mockChannel());

export const action = (actionName) => jest.fn();
const fusionContext = withFusionContext();
export const storiesOf = (groupName) => {
    const api = {
        decorators: [],
        parameters: {},

        add(storyName, story) {
            if (api.parameters.skipTest) {
                return;
            }

            describe(groupName, () => {
                it(storyName, () => {
                    // const decoratedStory = api.decorators.reduce((s, decorator) => {
                    //     try {
                    //         return decorator(s, { parameters: api.parameters });
                    //     } catch (e) {
                    //         return s;
                    //     }
                    // }, story);

                    const component = renderer.create(fusionContext(story));

                    // @TODO Enable custom names once released:
                    // > https://github.com/facebook/jest/pull/2094
                    expect(component.toJSON())
                        .toMatchSnapshot
                        // `${groupName}.${storyName}`
                        ();
                });
            });

            return api;
        },

        addDecorator() {
            return api;
        },

        addParameters(parameters) {
            api.parameters = { ...api.parameters, ...parameters };
            return api;
        },
    };

    return api;
};

export const addDecorator = () => {};
