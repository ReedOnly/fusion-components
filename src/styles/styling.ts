export default {
    grid: (value: number) => `calc(var(--grid-unit) * ${value}px)`,

    // Colors
    cssColors: {
        primary: 'var(--color-primary)',
        primaryAlt1: 'var(--color-primary-alt1)',
        primaryAlt2: 'var(--color-primary-alt2)',
        primaryAlt3: 'var(--color-primary-alt3)',
        primaryAlt4: 'var(--color-primary-alt4)',
        primaryAccent: 'var(--color-primary-accent)',
        primaryHover: 'var(--color-primary-hover)',
        primaryHoverAlt1: 'var(--color-primary-hover-alt1)',
        secondary: 'var(--color-secondary)',
        secondaryAlt1: 'var(--color-secondary-alt1)',
        secondaryAlt2: 'var(--color-secondary-alt2)',
        secondaryAlt3: 'var(--color-secondary-alt3)',
        secondaryAlt4: 'var(--color-secondary-alt4)',
        secondaryAccent: 'var(--color-secondary-accent)',
        black: 'var(--color-black)',
        blackAlt1: 'var(--color-black-alt1)',
        blackAlt2: 'var(--color-black-alt2)',
        blackAlt3: 'var(--color-black-alt3)',
        blackAlt4: 'var(--color-black-alt4)',
        blackAlt5: 'var(--color-black-alt5)',
        white: 'var(--color-white)',
        supplementary: 'var(--color-supplementary)',
        supplementaryAlt1: 'var(--color-supplementary-alt1)',
        supplementaryAlt2: 'var(--color-supplementary-alt2)',
        yellow: 'var(--color-yellow)',
        orange: 'var(--color-orange)',
        red: 'var(--color-red)',
        green: 'var(--color-green)',
        purple: 'var(--color-purple)',
        blue: 'var(--color-blue)',
        turquoise: 'var(--color-turquoise)',
    }
};
