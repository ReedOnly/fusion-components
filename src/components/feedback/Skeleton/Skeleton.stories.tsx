import React from 'react';
import { storiesOf } from '@storybook/react';
import withFusionStory from '../../../../.storybook/withFusionStory';

import { SkeletonBar, SkeletonButton, SkeletonDisc } from './index';
import { DiscSize } from './Disc';
import styling from 'styles/styling';

const sizes = {
    xlarge: 'xlarge',
    large: 'large',
    medium: 'medium',
    small: 'small',
};

const size = sizes.xlarge as DiscSize;

const stories = storiesOf('Feedback|Skeleton', module);
stories.addDecorator(withFusionStory('Skeleton'));

stories.add('Bar', () => (
    <>
        <div>
            <SkeletonBar />
        </div>
        <div>
            <SkeletonBar />
        </div>
        <div>
            <SkeletonBar />
        </div>
        <div>
            <SkeletonBar />
        </div>
    </>
));

stories.add('Button', () => (
    <>
        <div>
            <SkeletonButton width={styling.grid(10)} />
        </div>
        <div>
            <SkeletonButton width={styling.grid(10)} />
        </div>
        <div>
            <SkeletonButton width={styling.grid(10)} />
        </div>
        <div>
            <SkeletonButton width={styling.grid(10)} />
        </div>
    </>
));

stories.add('Disc', () => (
    <>
        <div>
            <SkeletonDisc size={size} />
        </div>
    </>
));
