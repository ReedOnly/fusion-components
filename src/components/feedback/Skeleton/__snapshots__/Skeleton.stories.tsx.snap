// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feedback|Skeleton Bar 1`] = `
<div
  className="container comfortable"
>
  <div
    className="container comfortable"
  >
    <div>
      <div
        className="bar comfortable skeleton"
        style={
          Object {
            "height": undefined,
            "width": "50%",
          }
        }
      />
    </div>
    <div>
      <div
        className="bar comfortable skeleton"
        style={
          Object {
            "height": undefined,
            "width": "50%",
          }
        }
      />
    </div>
    <div>
      <div
        className="bar comfortable skeleton"
        style={
          Object {
            "height": undefined,
            "width": "50%",
          }
        }
      />
    </div>
    <div>
      <div
        className="bar comfortable skeleton"
        style={
          Object {
            "height": undefined,
            "width": "50%",
          }
        }
      />
    </div>
  </div>
  <div
    className="overlay"
  />
  <div
    className="snacks"
  />
  <div
    className="dialog"
  />
</div>
`;

exports[`Feedback|Skeleton Button 1`] = `
<div
  className="container comfortable"
>
  <div
    className="container comfortable"
  >
    <div>
      <div
        className="skeleton comfortable button"
        style={
          Object {
            "width": "calc(var(--grid-unit) * 10px)",
          }
        }
      />
    </div>
    <div>
      <div
        className="skeleton comfortable button"
        style={
          Object {
            "width": "calc(var(--grid-unit) * 10px)",
          }
        }
      />
    </div>
    <div>
      <div
        className="skeleton comfortable button"
        style={
          Object {
            "width": "calc(var(--grid-unit) * 10px)",
          }
        }
      />
    </div>
    <div>
      <div
        className="skeleton comfortable button"
        style={
          Object {
            "width": "calc(var(--grid-unit) * 10px)",
          }
        }
      />
    </div>
  </div>
  <div
    className="overlay"
  />
  <div
    className="snacks"
  />
  <div
    className="dialog"
  />
</div>
`;

exports[`Feedback|Skeleton Disc 1`] = `
<div
  className="container comfortable"
>
  <div
    className="container comfortable"
  >
    <div>
      <div
        className="skeleton disc comfortable xlarge"
      />
    </div>
  </div>
  <div
    className="overlay"
  />
  <div
    className="snacks"
  />
  <div
    className="dialog"
  />
</div>
`;
